src/test/ui/rfc-2008-non-exhaustive/reachable-patterns.rs

```diff
@@ -0,0 +1,64 @@
+// Test that the reachable_patterns lint is triggered correctly.
+
+#![deny(reachable_patterns)]
+
+// aux-build:enums.rs
+extern crate enums;
+// aux-build:structs.rs
+extern crate structs;
+
+use enums::{EmptyNonExhaustiveEnum, NonExhaustiveEnum};
+use structs::{NormalStruct, UnitStruct, TupleStruct, FunctionalRecord};
+
+#[non_exhaustive]
+pub struct Foo {
+    a: u8,
+    b: usize,
+    c: String,
+}
+
+#[non_exhaustive]
+pub enum Bar {
+    A,
+    B,
+    C,
+}
+
+fn main() {
```

Comment:
Could you make sure you have test cases for enum variants too?

Reply:
I think there are right, you mean like this?
```rust
    match NonExhaustiveEnum::Unit {
        NonExhaustiveEnum::Unit => {},
        NonExhaustiveEnum::Tuple(_) => {},
        #[warn(reachable)]
        _ => {}
    }
```
Or do you mean something else?

Reply:
```rust
enum Foo {
  #[non_exhaustive] Bar { x: u32, y: u64 },
  Baz(u32, u16),
}

fn foo(f: Foo) {
  match f {
    Foo::Bar { 
      x: u32, 
      y: u64,
      #[warn(reachable)]
      ..
    } => {},
    Foo::Baz(..) => {},
  }
}
```

Reply:
Cool didn't know you could do that!